### PR TITLE
[diffusion] logging: log avail gpu mem while loading and generating

### PR DESCRIPTION
**Pull request by mickqian**
## Motivation

print some thing like:
```bash
[12-27 11:24:23] Loading vae from /root/.cache/huggingface/hub/models--Qwen--Qwen-Image-Edit-2511/snapshots/6f3ccc0b56e431dc6a0c2b2039706d7d26f22cb9/vae. avail mem: 34.97 GB
[12-27 11:24:23] Loaded vae: AutoencoderKLQwenImage from customized. avail mem: 34.97 GB, 0.47 GB consumed
[12-27 11:24:23] Memory usage of loaded modules: {'processor': 0.0, 'scheduler': 0.0, 'text_encoder': 15.45, 'tokenizer': 0.0, 'transformer': 38.05, 'vae': 0.47}

# generating...
# generation done

[12-27 11:24:53] Peak GPU memory: 46.76 GB, Remaining GPU memory at peak: 32.89 GB. Components that can stay resident: ['text_encoder', 'vae']
```

## TODO

add a server arg: `--dynamically-adjust-offload` to adopt the recommendation, based on history generations

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/platforms/cpu.py [hunk 1]
```diff
@@ -56,6 +59,31 @@ def get_current_memory_usage(
...
 
+    @classmethod
+    def get_available_gpu_memory(
+        cls,
+        device_id: int = 0,
+        distributed: bool = False,
+        empty_cache: bool = True,
+        cpu_group: Any = None,
+    ) -> float:
+        import psutil
```
Comment (BBuf) [inline python/sglang/multimodal_gen/runtime/platforms/cpu.py]: Can we remove import to head of the file.

File: python/sglang/multimodal_gen/runtime/platforms/cpu.py [hunk 2]
```diff
@@ -56,6 +59,31 @@ def get_current_memory_usage(
...
+
+        total_free_memory = psutil.virtual_memory().available
+        # For simplicity, we assume 1 NUMA node for now in this platform abstraction
+        # as get_cpu_ids_by_node is not available in multimodal_gen.runtime.utils
+        n_numa_node = 1
+        free_memory = total_free_memory / n_numa_node
+
+        if distributed:
+            import torch.distributed as dist
```
Comment (BBuf) [inline python/sglang/multimodal_gen/runtime/platforms/cpu.py:78]: Same as above.

File: python/sglang/multimodal_gen/runtime/loader/component_loader.py [hunk 3]
```diff
@@ -159,20 +185,27 @@ def load(
 
         if component is None:
             logger.warning("Loaded %s returned None", module_name)
+            consumed = 0.0
```
Comment (BBuf) [inline python/sglang/multimodal_gen/runtime/loader/component_loader.py:188]:
```suggestion
            consumed_mem = 0.0
```

Is better?

File: python/sglang/multimodal_gen/runtime/loader/component_loader.py [hunk 4]
```diff
@@ -159,20 +185,27 @@ def load(
...
         else:
+            current_gpu_mem = current_platform.get_available_gpu_memory()
+            consumed = get_memory_usage_of_component(component)
+            if consumed is None or consumed == 0.0:
+                consumed = gpu_mem_before_loading - current_gpu_mem
             logger.info(
-                f"Loaded %s: %s from: {source}",
+                f"Loaded %s: %s from {source}. avail mem: %.2f GB, %.2f GB consumed",
```
Comment (BBuf) [inline python/sglang/multimodal_gen/runtime/loader/component_loader.py:195]:
```suggestion
                f"Loaded %s: %s from {source}. GPU avail mem: %.2f GB, %.2f GB consumed",
```

**Review by BBuf — APPROVED** [review]

File: python/sglang/multimodal_gen/runtime/loader/component_loader.py (hunk 4 above)
Comment (mickqian) [reply python/sglang/multimodal_gen/runtime/loader/component_loader.py:195]: I'll leave it, for consistency with sglang srt

**Review by yhyang201 — APPROVED** [review]
